### ISSUE TITLE
Eval bug: Qwen 3.5 CPU bound on rocm / terrible performance

### ISSUE THREAD

**Issue by williamtwomey** [ggml-org/llama.cpp#20292]
### Name and Version

$ ./llama-cli --version
ggml_cuda_init: found 1 ROCm devices:
  Device 0: AMD Radeon AI PRO R9700, gfx1201 (0x1201), VMM: no, Wave Size: 32
load_backend: loaded ROCm backend from /home/will/git/llama.cpp-rocm/llama-b8235/libggml-hip.so
load_backend: loaded RPC backend from /home/will/git/llama.cpp-rocm/llama-b8235/libggml-rpc.so
load_backend: loaded CPU backend from /home/will/git/llama.cpp-rocm/llama-b8235/libggml-cpu-zen4.so
version: 8235 (ff52ee964)
built with GNU 11.4.0 for Linux x86_64

### Operating systems

Linux

### GGML backends

HIP

### Hardware

Ryzen 9600X + Radeon AI Pro R9700

### Models

Qwen 3.5 27B

### Problem description & steps to reproduce

Seeing 10x slowdowns on ROCM compared to Vulkan on Qwen 3.5 family models (other model famililes not impacted).

| Backend | Cold Prompt (REQ 1) | Cached (REQ 2) | Appended (REQ 3) |
|---------|-------------------|----------------|-------------------|
| ROCm    | 408.94s           | 27.23s         | 40.62s            |
| Vulkan  | 40.37s            | 12.61s         | 13.75s            |

llama-bench token generation (`-p 0 -n 128`) is similar on both backends (~21 t/s), so the issue is purely prompt processing.

Suspect this is a problem with Delta Net and the HIP runtime dispatching kernels one by one.

Profiled with `rocprof --stats` on pp128:

| Metric | Value |
|--------|-------|
| Total kernel dispatches | 14,977 |
| Total GPU compute time | **0.039 seconds** |
| Total wall time | **~4.1 seconds** |
| Time in dispatch overhead | **4.06 seconds (99%)** |
| Overhead per dispatch | ~271 µs |

Worth calling out GGML_HIP_GRAPHS=ON made things worse.

Also tested: `GPU_MAX_HW_QUEUES=8`, `HSA_ENABLE_SDMA=0`, `GGML_CUDA_ENABLE_GRAPHS=1`. None affected the dispatch overhead.

Steps to reproduce (have tested with and without FA and different batch sizes):

./llama-server --host 0.0.0.0 --port 8080 -m /models/Qwen3.5-27B.Q8_0.gguf -np 1 --batch-size 2048 --ubatch-size 512 -c 32000 -ngl 99 --flash-attn on

test.py snippet (from previous tests re: prompt re-processing)
```
#!/usr/bin/env python3
"""Qwen3.5 KV cache reuse reproducer."""

import requests, json, time, random, string

SERVER = "http://localhost:8080/v1/chat/completions"
NONCE = "".join(random.choices(string.ascii_lowercase, k=12))

filler = " ".join(
    f"The quick brown fox jumps over the lazy dog near the riverbank. "
    f"Paragraph {i} discusses distributed systems architecture including "
    f"consensus algorithms fault tolerance and horizontal scaling."
    for i in range(400)
)

messages = [
    {"role": "system", "content": f"You are a helpful assistant. [{NONCE}]"},
    {"role": "user", "content": filler},
]

print(f"Nonce: {NONCE}\n")

# REQ 1 — cold prompt
t0 = time.time()
response = requests.post(
    url=SERVER,
    headers={"Content-Type": "application/json"},
    data=json.dumps({"model": "test", "max_tokens": 200, "messages": messages}),
)
print(f"REQ 1 — cold prompt:    {time.time()-t0:.2f}s  (status {response.status_code})")
<snip>
```

Only happens on ROCM, Vulkan works fine.

Potentially related to https://github.com/ggml-org/llama.cpp/issues/20218

### Relevant log output

```llama_grammar_init_impl: failed to parse grammar
slot launch_slot_: id  0 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist 
slot launch_slot_: id  0 | task 0 | processing task, is_child = 0
slot update_slots: id  0 | task 0 | new prompt, n_ctx_slot = 32000, n_keep = 0, task.n_tokens = 14664
slot update_slots: id  0 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
slot update_slots: id  0 | task 0 | prompt processing progress, n_tokens = 2048, batch.n_tokens = 2048, progress = 0.139662
<Hangs here for several minutes>
```
